cli/iconBuilder.js

```diff
@@ -0,0 +1,167 @@
+const fs = require('fs'),
+	path = require('path');
+
+const imagePath = path.join(__dirname, '../components/icons/images');
+const outputPath = path.join(__dirname, '../.generated/icons');
+
+async function getSvgsInDir(dir) {
+	return new Promise((resolve, reject) => {
+		const dirPath = path.join(imagePath, dir);
+		fs.stat(dirPath, (err, stat) => {
+			if (err) return reject(err);
+			if (!stat.isDirectory()) return resolve([]);
+			fs.readdir(dirPath, (err, files) => {
+				if (err) return reject(err);
+				const svgs = files.filter((file) => {
+					return (path.extname(file) === '.svg');
+				}).map((file) => {
+					return {category: dir, name: file.substr(0, file.length - 4)};
+				});
+				resolve(svgs);
+			});
+		});
+	});
+}
+
+async function getSvgs() {
+	return new Promise(async(resolve, reject) => {
+		fs.readdir(imagePath, async(err, files) => {
+			if (err) return reject(err);
+			let svgs = [];
+			for (let i = 0; i < files.length; i++) {
+				const newSvgs = await getSvgsInDir(files[i]);
+				svgs = svgs.concat(newSvgs);
+				if (i === files.length - 1) {
+					resolve(svgs);
+				}
+			}
+		});
+	});
+}
+
+function clearDirectoryEntry(entryPath) {
+	return new Promise(async(resolve, reject) => {
+		fs.stat(entryPath, async(err, stat) => {
+			if (err) return reject(err);
+			if (stat.isDirectory()) {
+				await clearDirectory(entryPath);
+				fs.rmdir(entryPath, err => {
+					if (err) reject(err);
+					resolve();
+				});
+			} else {
+				fs.unlink(entryPath, err => {
+					if (err) reject(err);
+					resolve();
+				});
+			}
+		});
+	});
+}
+
+async function clearDirectory(dirPath) {
+	return new Promise(async(resolve, reject) => {
+		fs.readdir(dirPath, async(err, files) => {
+			if (err) return reject(err);
+			const promises = [];
+			for (const file of files) {
+				const newPath = path.join(dirPath, file);
+				promises.push(clearDirectoryEntry(newPath));
+			}
+			resolve(Promise.all(promises));
+		});
+	});
+}
+
+async function createCategory(category) {
```

Comment:
I won't mention the sync versions anymore... haha... you can decide what you like.  ;)  This one would become:
```
function createCategory(category) {
	const categoryPath = path.join(outputPath, category);
	fs.mkdirSync(categoryPath);
}
```